vispy/scene/canvas.py

```diff
@@ -207,6 +207,7 @@ def draw_visual(self, visual, event=None, viewport=None):
             Optionally specifies the viewport to use. If None, the entire
             physical size is used.
         """
+        self.set_current()
```

Comment:
Context switches are slow, but requesting a context that is already current is free (at least on my machine). I checked examples/basics/scene/line_update; ~630fps either way. We need a standard set of benchmarks...